src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -434,6 +435,21 @@ module.exports = function<T, P, I, TI, C, CX>(
       case HostComponent: {
         const instance : I = finishedWork.stateNode;
         attachRef(current, finishedWork, instance);
+
+        // Renderers may schedule work to be done after host components are mounted
+        // (eg DOM renderer may schedule auto-focus for inputs and form controls).
+        // These effects should only be committed when components are first mounted,
+        // aka when there is no current/alternate.
+        if (
+          !current &&
+          finishedWork.effectTag & Update
```

Comment:
Because it also gets called if there's a `Callback` effect, which host components don't have. I guess it's a bit weird either way, so I'm fine with keeping this as-is.